src/inmanta/execute/runtime.py

```diff
@@ -1015,8 +1015,8 @@ def execute(self) -> None:
         try:
             self._unsafe_execute()
         except RuntimeException as e:
-            e.set_statement(self.owner)
-            e.location = self.owner.location
+            replace: bool = isinstance(e, OptionalValueException)
```

Comment:
Special case for [test_optional_unset](https://github.com/inmanta/inmanta-core/blob/5c955d1278fe8d90440e598aa075526225adcce7/tests/compiler/test_relations.py#L711) 
Without this, the constructor would be reported.

Reply:
I fixed this differently